src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -88,33 +88,37 @@ function complement(V::AbstractAlgebra.Generic.FreeModule{T}, W::AbstractAlgebra
    return sub(V,_gens)
 end
 
-"""
-    permutation_matrix(F::Ring, Q::AbstractVector{T}) where T <: Int
-    permutation_matrix(F::Ring, p::PermGroupElem)
+@doc raw"""
+    permutation_matrix(R::NCRing, Q::AbstractVector{T}) where T <: Int
+    permutation_matrix(R::NCRing, p::PermGroupElem)
 
 Return the permutation matrix over the ring `R` corresponding to the sequence `Q` or to the permutation `p`.
 If `Q` is a sequence, then `Q` must contain exactly once every integer from 1 to some `n`.
 
 # Examples
 ```jldoctest
-julia> s = perm([3,1,2])
+julia> s = perm([3, 1, 2])
 (1,3,2)
 
-julia> permutation_matrix(QQ,s)
+julia> permutation_matrix(QQ, s)
 [0   0   1]
 [1   0   0]
 [0   1   0]
 
 ```
 """
-function permutation_matrix(F::Ring, Q::AbstractVector{<:IntegerUnion})
-   @assert Set(Q)==Set(1:length(Q)) "Invalid input"
-   Z = zero_matrix(F,length(Q),length(Q))
-   for i in 1:length(Q) Z[i,Q[i]] = 1 end
-   return Z
+function permutation_matrix(R::NCRing, Q::AbstractVector{<:IntegerUnion}; check::Bool = true)
+  if check
+    @assert Set(Q) == Set(1:length(Q)) "$Q must be a permutation"
+  end
+  Z = zero_matrix(R, length(Q), length(Q))
+  for i in 1:length(Q)
+    Z[i, Q[i]] = one(R)
```

Comment:
Do we have *any* example where this is faster than
```suggestion
    Z[i, Q[i]] = 1
```
If not I'd rather see us leave it as is, as was said: changing this is out of scope for this PR (so is insert `@doc raw`... but I don't mind it *shrug*)

Reply:
Let's revert it and I'll do some proper benchmark later.

Reply:
No, I don't have an example where it is faster with `one(R)` instead of `1`. At least in terms of allocations, it should usually not matter. And in general the performance of this function was so far not an issue for me.